src/modules/cl/kernel/non_linear_blend.cl

```diff
@@ -0,0 +1,57 @@
+#define saturate_8u(value) ((value) > 255 ? 255 : ((value) < 0 ? 0 : (value)))
+
+float gaussian(int x, int y, float std_dev) {
+  float res, pi = 3.14;
+  res = 1 / (2 * pi * stdDev * std_dev);
```

Comment:
this can be constant for all the threads, and hence better to precompute and pass it to kernel